gemini/use-cases/document-rag-on-llamaindex/llamaindex_rag.ipynb

```diff
@@ -0,0 +1,1458 @@
+        "    class.\n",
+        "    \"\"\"\n",
+        "\n",
+        "    storage_client = storage.Client()\n",
+        "\n",
+        "    # Searching for existing GCS bucket\n",
+        "    for bucket in storage_client.list_buckets():\n",
+        "        if bucket.name == bucket_name:\n",
+        "            print(f\"GCS Bucket {bucket_name} exists already in resource.\")\n",
+        "            return bucket\n",
+        "\n",
+        "    # Creating new bucket\n",
+        "    bucket = storage_client.bucket(bucket_name)\n",
+        "    bucket.storage_class = \"STANDARD\"\n",
+        "    new_bucket = storage_client.create_bucket(bucket, location=REGION)\n",
+        "\n",
+        "    print(\n",
+        "        \"Created bucket {} in {} with storage class {}\".format(\n",
+        "            new_bucket.name, new_bucket.location, new_bucket.storage_class\n",
+        "        )\n",
+        "    )\n",
+        "\n",
+        "    return new_bucket\n",
+        "\n",
+        "\n",
+        "def create_vertex_ai_search_index(index_name, index_dimensions):\n",
+        "    \"\"\"\n",
+        "    Creates a Vertex AI Search Index\n",
+        "    NOTE : This operation can take upto 30 minutes\n",
+        "    \"\"\"\n",
+        "\n",
+        "    # check if index exists\n",
+        "    index_names = [\n",
+        "        index.resource_name\n",
+        "        for index in aiplatform.MatchingEngineIndex.list(\n",
+        "            filter=f\"display_name={index_name}\"\n",
+        "        )\n",
+        "    ]\n",
+        "\n",
+        "    if len(index_names) == 0:\n",
+        "        print(f\"Creating Vector Search index {index_name} ...\")\n",
+        "        vs_index = aiplatform.MatchingEngineIndex.create_tree_ah_index(\n",
+        "            display_name=index_name,\n",
+        "            dimensions=index_dimensions,\n",
+        "            distance_measure_type=\"DOT_PRODUCT_DISTANCE\",\n",
+        "            shard_size=\"SHARD_SIZE_SMALL\",\n",
+        "            index_update_method=\"STREAM_UPDATE\",  # allowed values BATCH_UPDATE , STREAM_UPDATE,\n",
+        "            approximate_neighbors_count=5,\n",
+        "        )\n",
+        "        print(\n",
+        "            f\"Vector Search index {vs_index.display_name} created with resource name {vs_index.resource_name}\"\n",
+        "        )\n",
+        "    else:\n",
+        "        vs_index = aiplatform.MatchingEngineIndex(index_name=index_names[0])\n",
+        "        print(\n",
+        "            f\"Vector Search index {vs_index.display_name} exists with resource name {vs_index.resource_name}\"\n",
+        "        )\n",
+        "\n",
+        "    return vs_index\n",
+        "\n",
+        "\n",
+        "def create_vertexai_search_endpoint(endpoint_name):\n",
+        "    \"\"\"\n",
+        "    Creates a Vertex AI Search endpoint.\n",
```

Comment:
Change this to say "vector search". "Vertex AI Search" is a different product.

Reply:
Fixed.